java.lang.NoClassDefFoundError: Failed resolution of: Landroidx/appcompat/R$drawable;

App getting crash with below logs ,  Our app using Android Gradle version 3.6+ , archinamon version 4.1.0 and distributionUrl=https\://services.gradle.org/distributions/gradle-6.1-all.zip.


2020-04-06 10:09:27.677 20473-20473/com.example.smapireferenceapp E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.example.smapireferenceapp, PID: 20473
    java.lang.NoClassDefFoundError: Failed resolution of: Landroidx/appcompat/R$drawable;
        at androidx.appcompat.widget.AppCompatDrawableManager$1.<init>(AppCompatDrawableManager.java:63)
        at androidx.appcompat.widget.AppCompatDrawableManager.preload(AppCompatDrawableManager.java:57)
        at androidx.appcompat.app.AppCompatDelegateImpl.<init>(AppCompatDelegateImpl.java:328)
        at androidx.appcompat.app.AppCompatDelegateImpl.<init>(AppCompatDelegateImpl.java:278)
        at androidx.appcompat.app.AppCompatDelegate.create(AppCompatDelegate.java:221)
        at androidx.appcompat.app.AppCompatActivity.getDelegate(AppCompatActivity.java:543)
        at androidx.appcompat.app.AppCompatActivity.attachBaseContext(AppCompatActivity.java:98)
        at android.app.Activity.attach(Activity.java:7051)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2873)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3048)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1808)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "androidx.appcompat.R$drawable" on path: DexPathList[[zip file "/data/app/com.example.smapireferenceapp-7ftx-1zg3Oz8UpM7sw_eww==/base.apk"],nativeLibraryDirectories=[/data/app/com.example.smapireferenceapp-7ftx-1zg3Oz8UpM7sw_eww==/lib/x86, /system/lib]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:134)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        at androidx.appcompat.widget.AppCompatDrawableManager$1.<init>(AppCompatDrawableManager.java:63) 
        at androidx.appcompat.widget.AppCompatDrawableManager.preload(AppCompatDrawableManager.java:57) 
        at androidx.appcompat.app.AppCompatDelegateImpl.<init>(AppCompatDelegateImpl.java:328) 
        at androidx.appcompat.app.AppCompatDelegateImpl.<init>(AppCompatDelegateImpl.java:278) 
        at androidx.appcompat.app.AppCompatDelegate.create(AppCompatDelegate.java:221) 
        at androidx.appcompat.app.AppCompatActivity.getDelegate(AppCompatActivity.java:543) 
        at androidx.appcompat.app.AppCompatActivity.attachBaseContext(AppCompatActivity.java:98) 
        at android.app.Activity.attach(Activity.java:7051) 
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2873) 
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3048) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1808) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6669) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 
2020-04-06 10:09:27.719 20473-20473/com.example.smapireferenceapp I/Process: Sending signal. PID: 20473 SIG: 9

--- a/android-gradle-aspectj/src/main/kotlin/com/archinamon/api/transform/AspectJTransform.kt
+++ b/android-gradle-aspectj/src/main/kotlin/com/archinamon/api/transform/AspectJTransform.kt
@@ -240,6 +240,16 @@ internal abstract class AspectJTransform(val project: Project, private val polic
             //copy compiled .class files to output directory
             FileUtil.copyDir(compiledAj, outputDir)
         }
+
+        transformInvocation.inputs.forEach { transformInput ->
+            // Ensure JARs are copied as well:
+            transformInput.jarInputs.forEach {
+                it.file.copyTo(
+                    transformInvocation.outputProvider.getContentLocation(it.name, inputTypes, scopes, Format.JAR),
+                    overwrite = true
+                )
+            }
+        }
     }
 
     private fun copyJar(outputProvider: TransformOutputProvider, jarInput: JarInput?): Boolean {

